CI: Release update and screenshot gathering automation

kz6fittycent opened the pull request:
- Modified script for getting latest release from GH API
- Minor cleanup of some old comments in yml files
- Update snapraft.yaml to `version` instead of `adopt-info`

I searched for anything that may be lacking, but let me know if/what I missed.

kz6fittycent committed:
Update snapcraft.yaml

just in case!

kz6fittycent committed:
Update snapcraft.yaml

added cleanup

kz6fittycent committed:
Update snapcraft.yaml

kz6fittycent committed:
Update snapcraft.yaml

kz6fittycent committed:
updated yaml and icon

kz6fittycent committed:
Node16.x

Testing

kz6fittycent committed:
Initial test for CI workflow

Copied with minor mods from Discord snap
- snap promote candidate and stable
- sync version w/ upstream (not sure about this one)

kz6fittycent committed:
CI Workflow - redo

Restored to original from Discord
Cleanup of test-snap-can-build.yaml

kz6fittycent committed:
Merge branch 'master' into testing

kz6fittycent committed:
Merge pull request #7 from kz6fittycent/testing

CI: add release update and screenshot gathering automation

kz6fittycent committed:
Remove build-trigger file

kz6fittycent committed:
adjusted script for jenkins latest version

kz6fittycent committed:
Forgot to change snapcraft.yaml

adopt-info -> version

merlijn-sebrechts reviewed (changes requested):
I see some of these actions are still using the old in-repo ones. Can you take a look at https://github.com/snapcrafters/signal-desktop/tree/candidate/.github/workflows and update accordingly?

I apologize for this churn. The current ones in signal-desktop should be the final ones, I promise!

kz6fittycent commented:
> I see some of these actions are still using the old in-repo ones. Can you take a look at https://github.com/snapcrafters/signal-desktop/tree/candidate/.github/workflows and update accordingly?
> 
> I apologize for this churn. The current ones in signal-desktop should be the final ones, I promise!

No problem! I was fully expecting to redo something because of my own errors anyway. Give me a couple of days to get to it.

kz6fittycent committed:
Updated workflows to match Signal snap

- Updated/renamed
  - update-to-stable
  - pull-request
  - release-to-candidate
  - sync-version-with-upstream
Removed:
  - Removed test-snap-can-build

kz6fittycent committed:
Forgot quotes around version # in snap yaml

kz6fittycent committed:
updated  sync-version-with-upstream

Needed to change some of the script to allow the double-quotes to remain around the version for the snap.

merlijn-sebrechts commented on .github/vmctl:
```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
```
You can remove this entire file. The new workflow downloads the latest version automatically

kz6fittycent replied:
stand by

merlijn-sebrechts reviewed (changes requested):
A small change. Otherwise, it looks good!

merlijn-sebrechts commented:
I've added the tokens, so this should work as soon as we merge it!

kz6fittycent committed:
rm vmctl

https://github.com/snapcrafters/jenkins/pull/41#discussion_r1414405981

merlijn-sebrechts reviewed (approved):
Guddd!
